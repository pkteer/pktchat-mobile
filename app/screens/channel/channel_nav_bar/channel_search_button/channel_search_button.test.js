// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {TouchableOpacity} from 'react-native';

import Preferences from '@mm-redux/constants/preferences';
import {shallowWithIntl} from '@test/intl-test-helper';

import ChannelSearchButton from './channel_search_button';

describe('ChannelSearchButton', () => {
    const baseProps = {
        actions: {clearSearch: jest.fn()},
        theme: Preferences.THEMES.denim,
    };

    test('should match, full snapshot', () => {
        const wrapper = shallowWithIntl(
            <ChannelSearchButton {...baseProps}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper.find(TouchableOpacity).first().exists()).toEqual(true);
    });

    test('Should be accessible', () => {
        const wrapper = shallowWithIntl(
            <ChannelSearchButton {...baseProps}/>,
        );
        expect(wrapper.find(TouchableOpacity).first().props().accessible).toBeTruthy();
    });

    test('Should have the correct accessibilityHint', () => {
        const wrapper = shallowWithIntl(
            <ChannelSearchButton {...baseProps}/>,
        );
        expect(wrapper.find(TouchableOpacity).first().props().accessibilityHint).toEqual('Opens the channel search modal');
    });

    test('Should have the correct accessibilityLabel', () => {
        const wrapper = shallowWithIntl(
            <ChannelSearchButton {...baseProps}/>,
        );
        expect(wrapper.find(TouchableOpacity).first().props().accessibilityLabel).toEqual('Search');
    });
});
