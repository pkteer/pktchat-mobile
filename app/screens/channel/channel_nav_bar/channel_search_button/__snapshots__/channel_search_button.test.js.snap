// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelSearchButton should match, full snapshot 1`] = `
<View
  style={
    Object {
      "width": 40,
    }
  }
>
  <TouchableOpacity
    accessibilityHint="Opens the channel search modal"
    accessibilityLabel="Search"
    accessibilityRole="button"
    accessible={true}
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="channel.search.button"
  >
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "position": "relative",
        }
      }
    >
      <CompassIcon
        name="magnify"
        size={24}
        style={
          Object {
            "backgroundColor": "#192a4d",
            "color": "#ffffff",
          }
        }
      />
    </View>
  </TouchableOpacity>
</View>
`;
